prov/gni/src/gnix_msg.c

```diff
@@ -120,22 +133,47 @@ static inline int __gnix_recv_completion(struct gnix_fid_ep *ep,
 			req->msg.tag);
 }
 
+static int __gnix_send_err(struct gnix_fid_ep *ep, struct gnix_fab_req *req)
+{
+	uint64_t flags = FI_SEND | FI_MSG;
+	int rc;
+
+	flags |= req->msg.send_flags & FI_TAGGED;
+
+	if (ep->send_cq) {
+		rc = _gnix_cq_add_error(ep->send_cq, req->user_context,
+					flags, 0, 0, 0, 0, 0, FI_ECANCELED,
+					GNI_RC_TRANSACTION_ERROR, NULL);
+		if (rc != FI_SUCCESS)  {
+			GNIX_WARN((FI_LOG_CQ | FI_LOG_EP_DATA),
```

Comment:
This is going to lead to an out of bounds array access.  The first parameter is an index (enum) into an array of strings used to print the error message, so these are not flags that can be OR'ed.